runtime/autoload/lsp/util.vim

```diff
@@ -0,0 +1,17 @@
+
+""
+" A function for finding the base directory containing a file
+function! lsp#util#find_root_uri(pattern) abort
```

Comment:
How about adding this to `eval.c` ? Or at least, keep it private until then. We should not be creating a VimL utility library separate from `nvim` itself--this doesn't make sense for users.

Reply:
I just removed it for now. I think people can write this or already have a plugin that does this for them if they're using a "project" based language like this.